library/java/src/io/envoyproxy/envoymobile/engine/EnvoyConfiguration.java

```diff
@@ -1,11 +1,64 @@
 package io.envoyproxy.envoymobile.engine;
 
-public interface EnvoyConfiguration {
+public class EnvoyConfiguration {
+
+  public final String configYAML;
+  public final Integer connectTimeoutSeconds;
+  public final Integer dnsRefreshSeconds;
+  public final Integer statsFlushSeconds;
+
+  /**
+   * Create an EnvoyConfiguration with a user provided configuration YAML.
+   *
+   * @param configYAML configuration to be used for starting Envoy.
+   */
+  public EnvoyConfiguration(String configYAML) {
```

Comment:
This should be removed since the string should no longer be stored within the config